### ISSUE TITLE
Migrate away from react-infinite-calendar

### ISSUE THREAD

**Issue by astone123** [cypress-io/cypress-realworld-app#1419]
The [react-infinite-calendar](https://github.com/clauderic/react-infinite-calendar) package has been unmaintained for years and doesn't support React 18. We should migrate to a different package for our calendar UI.

It looks like the only place we use this is inside of `TransactionDateRangeFilter.tsx`. I'm sure we could find a suitable replacement

**Comment by MikeMcC399** [comment]
See installation warnings for `react-infinite-calendar`. The repo is already using the latest version, which supports only up to `react@15` (considering also its dependencies):

```text
$ yarn
yarn install v1.22.22
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning " > react-infinite-calendar@2.3.1" has incorrect peer dependency "react@^15.3.0 || ^16.0.0-alpha".
warning "react-infinite-calendar > react-tiny-virtual-list@2.2.0" has incorrect peer dependency "react@15.x || 16.x".
warning "react-infinite-calendar > react-transition-group@1.2.1" has incorrect peer dependency "react@^15.0.0 || ^16.0.0".
warning "react-infinite-calendar > react-transition-group@1.2.1" has incorrect peer dependency "react-dom@^15.0.0 || ^16.0.0".
warning "react-infinite-calendar > recompose@0.22.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0".
warning " > @cypress/code-coverage@3.12.44" has unmet peer dependency "@babel/preset-env@^7.0.0".
warning " > @cypress/code-coverage@3.12.44" has unmet peer dependency "babel-loader@^8.3 || ^9".
warning "@cypress/code-coverage > @cypress/webpack-preprocessor@6.0.2" has unmet peer dependency "@babel/preset-env@^7.0.0".
warning "@cypress/code-coverage > @cypress/webpack-preprocessor@6.0.2" has unmet peer dependency "babel-loader@^8.3 || ^9".
warning Workspaces can only be enabled in private projects.
warning Workspaces can only be enabled in private projects.
[5/5] Building fresh packages...
$ husky install && patch-package
husky - Git hooks installed
patch-package 7.0.2
Applying patches...
react-virtualized@9.22.5 ✔
```

**Comment by shiv343** [comment]
Is the issue still open ? i would like to work on this issue

**Comment by babyitachi** [comment]
@MikeMcC399 shall we update React version for this?

**Comment by MikeMcC399** [comment]
@babyitachi 

> shall we update React version for this?

I don't believe that it is necessary to update `react` from its current version [react@18.2.0](https://github.com/facebook/react/releases/tag/v18.2.0) in this repo in order to replace the calendar component. Cypress continues to list React 18 - 19 under [Supported Frameworks](https://docs.cypress.io/app/component-testing/get-started#Supported-Frameworks) and does not require an update to React 19.

The first task would be to replace the calendar. Afterwards, it may be possible to update React to v19.

---

The community contributor @shiv343 made an attempt in PR https://github.com/cypress-io/cypress-realworld-app/pull/1621 to resolve this issue, however the task was not completed, because the Cypress tests were not updated accordingly, and finally the PR was closed.

It may be possible to revive the contents of PR https://github.com/cypress-io/cypress-realworld-app/pull/1621 by copying it and adding tests to it.

The PR attempted to migrate from [react-infinite-calendar@2.3.1](https://www.npmjs.com/package/react-infinite-calendar/v/2.3.1) to [react-calendar@5.1.0](https://github.com/wojtekmaj/react-calendar/releases/tag/v5.1.0) keeping the version [react@18.2.0](https://github.com/facebook/react/releases/tag/v18.2.0) the same.

[react-infinite-calendar@2.3.1](https://www.npmjs.com/package/react-infinite-calendar/v/2.3.1) is the latest available version, published 8 years ago and it is compatible only with `"react": "^15.3.0 || ^16.0.0-alpha"`. This is not compatible with this repo.

The latest version [react-calendar@6.0.0](https://github.com/wojtekmaj/react-calendar/releases/tag/v6.0.0) is compatible with `"react": "^16.8.0 || ^17.0.0 || ^18.0.0 || ^19.0.0"`.

**Comment by babyitachi** [comment]
Hey @MikeMcC399,
I have successfully migrated react-infinite-calendar@2.3.1 to react-calendar@5.1.0. 
Here is the #1650 PR for the same.

Please review at your convenience. Thanks!
